package/kedro_viz/models/flowchart.py

```diff
@@ -93,22 +108,29 @@ class GraphNode(abc.ABC):
     type: str
 
     # the underlying Kedro object for each graph node, if any
-    kedro_obj: Optional[Union[KedroNode, AbstractDataset]] = field(default=None)
+    kedro_obj: Optional[Union[KedroNode, AbstractDataset]]
 
     # the tags associated with this node
-    tags: Set[str] = field(default_factory=set)
+    tags: Set[str] = set()
 
     # the set of registered pipeline IDs this node belongs to
-    pipelines: Set[str] = field(default_factory=set)
+    pipelines: Set[str] = set()
 
     # The original namespace on this node.
     # N.B.: in Kedro, modular pipeline is implemented by declaring namespace on a node.
     # For example, node(func, namespace="uk.de") means this node belongs
     # to the modular pipeline "uk" and "uk.de"
-    namespace: Optional[str] = field(init=False, default=None)
```

Comment:
Noted that it used to be not possible to be initialised (it is possible now). I am not sure how this may affect downstream after many layers of propagation. Is it possible to maintain this behavior using the validation. see https://github.com/pydantic/pydantic/issues/2019